src/MCrow.cpp

```diff
@@ -80,14 +80,93 @@ mcrow_acceptance(const double full_ratio)
   }
 }
 
+Eigen::VectorXd
+mcrow_make_yi_star(const double stepsize,
+                   const Eigen::VectorXd& Yi)
+{
+  VectorXd Yi_star(Yi.size());
+
+  // This class calls GetRNGState in it's constructor and PutRNGState
+  // in its destructure, which update .Random.seed properly.  Since we
+  // don't export this function, we must do this manually.
+  Rcpp::RNGScope rcpp_rngScope_gen;
+
+  for (int i = 0; i < Yi_star.size(); ++i) {
```

Comment:
It's definitely possible it's a bit slower than just calculating the rnorm vals in R.  Mainly it is to avoid precomputing a big matrix of them in R then passing that in to Rcpp.  Good thing is that the rnorm calculation from Rcpp wasn't a bottleneck when doing the profiling.